### ISSUE TITLE
Marker custom icons

### ISSUE THREAD

**Issue by macropp** [ng2-ui/ng2-map#96]
**Setup**
I have a map laid out on my page and I’m displaying markers on it successfully. I want the markers to have custom graphics (which I’m setting using the icon @input).

The problem is that it works if set the icon to a URL string, but I’m unable to make it work with size and scaledSize properties set. It also works if I set icon @input as an object with only url key in it.

I’m trying to set the custom icon properties using this object:

```javascript
markerIcon = {
 size: { width: 32, height: 32 },
 scaledSize: { width: 32, height: 32 },
 anchor: { x: 16, y: 16 },
 url: '/marker.svg'
};
```

I also tried waiting for the map's ready event and then assigning size and scaledSize properties of the markerIcon object to ```new google.maps.Size(32, 32)``` and anchor property to ```new google.maps.Point(16,16)```, but the output is the same.

[Plunker example](https://plnkr.co/edit/BVocDNQOC8V9UKFPnxzt?p=preview)

**Current behavior**

  - Marker is not displayed at all

**Expected/desired behavior**

  - Marker should be displayed according to the provided size and image.

**Other information**

I observed that if you assign the same values in marker initialize event handler to ```event.icon``` property, marker will be displayed correctly, but as soon as you zoom, incorrect part of the marker icon is displayed.

Funny thing is that, this hack works fine even when zooming on Plunker, but does not in my case. Only difference I see is that Plunker has a PNG file in the url and I use SVG.

**Comment by allenhwkim** [comment]
It seems working for me with `[icon]` without any modification.

https://plnkr.co/edit/qu56sf?p=preview

```
    <ng2-map zoom="14" center="Brampton, Canada">
      <marker position="Brampton, Canada"
        [icon]="{
          url: 'https://plnkr.co/img/plunker.png',
          anchor: [16,16],
          size: [32,32],
          scaledSize: [32,32]
        }">
      </marker>
    </ng2-map>
```

**Comment by macropp** [comment]
@allenhwkim sorry for the late reply. Seems to be working with arrays as anchor, size and scaledSize values. I was trying with JS Objects for some reason..

Thanks for the clarification.

**Comment by allenhwkim** [comment]
@macropp One thing to remember is that "google.maps" is not available until `ng2-map` is initialized unless you include google maps api separately into index.html.

Thus, `ng2-map` accepts google objects parameters as string or as a simple hash and make a proper initialization internally, which makes user little bit confused, but also having some convenience.

Let me know if I can improve more from yoru experience.

**Comment by macropp** [comment]
@allenhwkim, yeah eventually I also figured that google namespace is only available after map's ready event is dispatched. 

Thanks for your help, provided method is pretty convenient.
